webviz_subsurface/plugins/_tornado_plotter_fmu/shared_settings/_filters.py

```diff
@@ -0,0 +1,154 @@
+import json
+from typing import List, Optional
+
+import webviz_core_components as wcc
+from dash import ALL, Input, Output, callback, callback_context
+from dash.development.base_component import Component
+from webviz_config.webviz_plugin_subclasses import LayoutUniqueId, SettingsGroupABC
+
+from webviz_subsurface._providers import EnsembleTableProvider
+
+from .._plugin_ids import PlugInIDs
+
+
+class Filters(SettingsGroupABC):
+    """Settingsgroup for filters"""
+
+    class IDs:
+        # pylint: disable=too-few-public-methods
+        SINGLE_FILTER = "single-filter"
+        MULTI_FILTER = "multi-filter"
+
+    def __init__(
+        self,
+        table_provider: EnsembleTableProvider,
+        single_filters: List[str],
+        multi_filters: List[str],
+        ensemble_name: str,
+    ) -> None:
+        super().__init__("Filters")
+        self._table_provider = table_provider
+        self._single_filters = single_filters
+        self._multi_filters = multi_filters
+        self._ensemble_name = ensemble_name
+        self._plugin_unique_id = LayoutUniqueId(plugin_uuid=PlugInIDs.PlugIn.PLUGIN_ID)
+
+    def layout(self) -> List[Component]:
+        elements = []
+
+        # Creates dropdowns for any data columns added as single value filters
+        for selector in self._single_filters:
+            values = self._table_provider.get_column_data([selector])[selector].unique()
+
+            elements.append(
+                wcc.Dropdown(
+                    label=selector,
+                    id={
+                        "id": self.uuid(Filters.IDs.SINGLE_FILTER),
```

Comment:
you can set it a constant outside the loop. As Erike did in line 31in
https://github.com/EirikPeirik/webviz-subsurface/blob/WellAnalysis_trail/webviz_subsurface/plugins/_well_analysis/_views/_settings.py

Reply:
same as line 64

Reply:
Good idea! Did not think of that, added the change:)